docs/design/training-pre-check.md

```diff
@@ -0,0 +1,121 @@
+# Training Pre Check Procedure Introduction
+
+## Background
+
+As the scale of training production expands, the factors that may cause 
+runtime exceptions related to the environment during training are gradually 
+increasing. Although the current DLRover adds a so-called node-check before 
+training (which runs a small-scale training task to detect any abnormal 
+scenarios), it primarily observes availability from the training dimension 
+perspective. Therefore, it is not easy to specifically diagnose problematic 
+nodes and classify issues. 
+
+So we are seeking a more advanced and precise check, 
+conducting certain related inspections right when all nodes are in place 
+(or not yet in place) to identify issues as early as possible and implement 
+fault tolerance or blockages in advance. The checks mentioned earlier rely more 
+on external services, as they require specialized and more precise validation 
+of components such as containers and networks. Traditional training frameworks 
+do not and should not involve these components. 
+
+Considering that the actual 
+checks are highly related to the specific deployment environment and heavily 
+rely on external capabilities, there are currently no actual check 
+implementations added on the open-source side, only the exposure of interfaces 
+and the implementation of processes. This article only discusses the core 
+design of this aspect.
+
+
+## Target
+
+- Support configurable multiple types of serial synchronous check before 
+  training starts. (primary)
+- Support limited checks during training execution. (secondary)
```

Comment:
1. It is unclear what is "limited checks".
2. Remove primary for the 1st one. If the second one is not mandatory, maybe label it with optional.

Reply:
1. Add more description.
2. Here, the 'primary' and 'secondary' refer to the targets, distinguishing between the primary target and the secondary target. It does not imply a mandatory and optional relationship.